web/src/message_view.js

```diff
@@ -152,6 +152,13 @@ function create_and_update_message_list(filter, id_info, opts) {
             });
         }
 
+        // In case we have some already fetched messages, we add them to
+        // the msg_data, and build the list around it.
+        if (opts.fetched_message_ids !== undefined) {
+            const msgs = opts.fetched_message_ids.map((msg_id) => message_store.get(msg_id));
+            msg_data.add_messages(msgs);
+        }
```

Comment:
How does this interact with `maybe_add_local_messages`? I don't see how it could be correct to be adding messages from those two sources without any care taken about whether the temporal regions of message history are adjacent. @amanagr can perhaps work with on how to get this correct.

Reply:
A lot of the new logic in `message_view` seems unnecessary, we should just be using `adjusted_terms_if_moved` logic for this. I don't think we need `get_narrow_data_for_with_operator`.

Reply:
>How does this interact with maybe_add_local_messages? I don't see how it could be correct to be adding messages from those two sources without any care taken about whether the temporal regions of message history are adjacent.

The section of code mentioned is only executed when `fetched_message_ids` is not undefined --> when the messages corresponding to the narrow are fetched from server using `message_fetch.load_messages_for_narrow`.  When this function fetches messages, it adds them to the `message_store`.

Reply:
@amanagr I tried to refactor `message_view` and removed the `get_narrow_data_for_with_operator`. could you check it out?

Also, I am not sure if `recent_view_data` is the best place for accessing the last message id of the topic when clicked on a topic from left sidebar -- since for certain topics, it was not available in `recent_view_data.get_conversations()` when clicked (for that topic `stream_id:topic` key was returning undefined).